superset-frontend/src/explore/components/controls/BoundsControl.jsx

```diff
@@ -45,6 +50,22 @@ export default class BoundsControl extends React.Component {
     this.onChange = this.onChange.bind(this);
     this.onMinChange = this.onMinChange.bind(this);
     this.onMaxChange = this.onMaxChange.bind(this);
+    this.update = this.update.bind(this);
+  }
+
+  componentDidUpdate(prevProps) {
+    if (!_.isEqual(prevProps.value, this.props.value)) {
+      this.update();
+    }
+  }
+
+  update() {
+    this.setState({
+      minMax: [
+        Number.isNaN(this.props.value[0]) ? '' : this.props.value[0],
+        Number.isNaN(this.props.value[1]) ? '' : this.props.value[1],
```

Comment:
This behaves a little funny in practice, in that if you type a non-numeric value into one of these fields, it doesn't complain, and the value sticks. However, that non-numeric value IS removed if you type into the _other_ field. 

In other words, you can enter a good value in `min` and then a bad value in `max` - that bad value is only removed if you go back and change `min`

Reply:
Replaced `Input` by `InputNumber`. Now it only allows numeric values and the user can click on the spinner.